### DISCUSSION TITLE
<x-jet-action-message on="event"> no longer working after migration from Livewire 2 to 3

### DISCUSSION THREAD

**Discussion by acharseth** [livewire/livewire#8025] [1. Help] [answered]
I am firing an event in a Livewire controller like this: `$this->dispatch('rcFound'); ` and catching it in a blade file like this:
```
<x-jet-action-message class="mr-3" on="rcFound">
                            {{ __('Found.') }}
                        </x-jet-action-message> 
```
In Livewire 2 this used to work fine. After migration to Livewire 3 I get the following warning and error message:
```
app.8c8ff37e.js:32 Alpine Expression Error: Cannot read properties of undefined (reading 'on')

Expression: "window.Livewire.find('4Oni8rhXJmOi6nCkfOph').on('rcFound', () => { clearTimeout(timeout); shown = true; timeout = setTimeout(() => { shown = false }, 2000);  })"
```
And corresponding error message (I think):
```
app.8c8ff37e.js:36 Uncaught TypeError: Cannot read properties of undefined (reading 'on')
    at [Alpine] window.Livewire.find('4Oni8rhXJmOi6nCkfOph').on('rcFound', () => { clearTimeout(timeout); shown = true; timeout = setTimeout(() => { shown = false }, 2000);  }) (eval at <anonymous> (http://localhost:8000/build/assets/app.8c8ff37e.js:36:670), <anonymous>:3:76)
```

Why isn't this working any longer in Livewire 3 and how do I fix this?

**Comment by danie-ramdhani** [comment]
hmm... try something like this:

```blade
<button x-on:click="$dispatch('rc-found')">
    Dispatch
</button>

<div x-on:rc-found.window="console.log($el)">
    Foo
</div>
```

**Reply by acharseth** [reply]
What is `$el`, @danie-ramdhani?

Upgraded to Jetstream 4.2 (see below) and did the following replacement:

From:
```
<x-action-message class="mr-3" on="rcFound"> 
    {{ __('Found.') }}
</x-action-message>
```

To: 
```
<div x-on:rc-found.window="console.log($el)">
     {{ __('Found.') }}
</div>
```

This removes the error message but I don't get the expected result, "Found" only displayed event triggered, now it always displays!?
Anyway I still don't understand why it does not work. I had high hopes that upgrading Jetstream would work, but no. :-(

**Reply by acharseth** [reply]
With your suggested code, @danie-ramdhani, I do get a log entry in the browser when I click "Dispatch" so the triggering and catching the event works (though I expected "Foo" only to show up after clicking "Dispatch").
Here is the event in the log:
```
[Log] <div x-on:rc-found.window="console.log($el)">
                                Foo
                            </div>
```

By the way the triggering and catching functionally now (after upgrading Jetstream) works with the original code as well, it is just that I get the error message in the console.

**Comment by acharseth** [comment]
I found out that I used an old version of Jetstream which was not compatible with Livewire 3 so I upgraded Jetstream to ver 4.2. 
Unfortunately this did not help. I still get the same error message in the browser console.

**Comment by acharseth** [comment, marked as answer]
Resolved by removing (commenting out) Alpine stuff from resources/app/app.js:
```
import './bootstrap';

//import Alpine from 'alpinejs';

//window.Alpine = Alpine;

//Alpine.start();
```
And then running `npm run build`
